apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -215,5 +217,129 @@ string ExtractPath(DocumentReference documentReference)
         internal IAsyncEnumerable<CollectionReference> ListCollectionsAsync(DocumentReference parent) =>
             Client.ListCollectionIdsAsync(parent?.Path ?? RootPath)
                 .Select(id => new CollectionReference(this, parent, id));
+
+        // TODO: Is it appropriate to use ConfigureAwait(false) here? Should the caller be able to specify the context for executing their callback?
+        // TODO: Collect up all the exceptions we've retried?
+        // TODO: Is it nice to have all these overloads?
```

Comment:
I think just keeping the two overloads with async callbacks is fine. As the `RunTransactionAsync` method is itself async, I think requiring an async callback is reasonable.

Reply:
I thought so until I started writing tests - but as the modification methods are synchronous, I think it won't be uncommon to have a synchronous callback, in which case you get a warning due to an async lambda with no awaits. (Or you have to return Task.FromResult directly.)